src/interface/swarm.cpp

```diff
@@ -317,28 +354,469 @@ void Swarm::Defrag() {
         }
       });
 
-  for (int m = 0; m < intVector_.size(); m++) {
-    auto &vec = intVector_[m]->Get();
-    pmb->par_for(
-        "Swarm::DefragInt", 0, max_active_index_, KOKKOS_LAMBDA(const int n) {
-          if (from_to_indices(n) >= 0) {
-            vec(from_to_indices(n)) = vec(n);
+  SwarmVariablePack<Real> vreal;
+  SwarmVariablePack<int> vint;
+  PackIndexMap rmap;
+  PackIndexMap imap;
+  PackAllVariables(vreal, vint, rmap, imap);
+  int real_vars_size = realVector_.size();
+  int int_vars_size = intVector_.size();
+
+  pmb->par_for(
+      "Swarm::DefragVariables", 0, max_active_index_, KOKKOS_LAMBDA(const int n) {
+        if (from_to_indices(n) >= 0) {
+          for (int i = 0; i < real_vars_size; i++) {
+            vreal(i, from_to_indices(n)) = vreal(i, n);
           }
-        });
+          for (int i = 0; i < int_vars_size; i++) {
+            vint(i, from_to_indices(n)) = vint(i, n);
+          }
+        }
+      });
+
+  // Update max_active_index_
+  max_active_index_ = num_active_ - 1;
+}
+
+void Swarm::SetupPersistentMPI() {
+  vbswarm->SetupPersistentMPI();
+
+  allreduce_request_ = MPI_REQUEST_NULL;
+
+  // Index into neighbor blocks
+  auto pmb = GetBlockPointer();
+  auto neighborIndices_h = neighborIndices_.GetHostMirror();
+
+  // Region belonging to this meshblock
+  // for (int k = 1; k < 3; k++) {
```

Comment:
The commented out for loop can you remove it, if you want to indicate the indices belonging to the meshblock I would simply say indices so and so without the code syntax.